File: java/collatz-conjecture/src/main/java/CollatzCalculator.java
class CollatzCalculator {

    int computeStepCount(int start) {
        
        if (start < 1) {
            throw new IllegalArgumentException("Only positive integers are allowed");
        }

        return start == 1 ? 0 : 1 + computeStepCount(start % 2 == 0 ? start / 2 : 3 * start + 1);
    }

}

File: kotlin/collatz-conjecture/src/main/kotlin/CollatzCalculator.kt
object CollatzCalculator {
    fun computeStepCount(start: Int): Int {

        require(start > 0)

        return when {
            start == 1 -> 0
            start % 2 == 0 -> 1 + computeStepCount(start / 2)
            else -> 1 + computeStepCount(3 * start + 1)
        }
    }
}
